crates/sui-core/src/batch_bls_verifier.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use shared_crypto::intent::{Intent, IntentScope};
+use sui_types::{
+    committee::Committee,
+    crypto::{AuthoritySignInfoTrait, VerificationObligation},
+    error::SuiResult,
+    message_envelope::Message,
+    messages::CertifiedTransaction,
+};
+
+use tap::TapFallible;
+use tracing::error;
+
+/// Verifies all certificates - if any fail return error.
+pub fn batch_verify_all_certificates(
+    committee: &Committee,
+    certs: &[CertifiedTransaction],
+) -> SuiResult {
+    // Verify user signatures
+    for cert in certs {
+        cert.data().verify(None)?;
+    }
+
+    batch_verify_certificates_impl(committee, certs)
+}
+
+/// Verifies certificates in batch mode, but returns a separate result for each cert.
+pub fn batch_verify_certificates(
```

Comment:
iiuc atm we just introduce the related functions (this one being the most important as it returns result per cert), but not use them in Sui yet, right? 

When we actually use/invoke it (this is part II?) we're going to apply some strategy based on:
 a) on high throughput: each batch will have a max cert number
 b) time bounded (we tolerate some latency, then batch)

Note that the total latency per cert will be: 
a) max waiting time until we form a batch 
+
b) batch verification time

Reply:
That's right - I just finished writing that part of it, and will put it up for review tomorrow after I write a benchmark for it.